### ISSUE TITLE
it's better to make optimizer as a class.

### ISSUE THREAD

**Issue by xpqiu** [fastnlp/fastNLP#17]
https://github.com/fastnlp/fastNLP/blob/7c2f260a2fdd91ff5c62e25a84f56deeed2b7483/fastNLP/action/optimizor.py#L1

**Comment by FengZiYjun** [comment]
We do not implement optimizers, only borrow them from pytorch. Currently a single function as an interface is enough. Making optimizer as a class is reasonable if it provides more functionalities.

**Comment by xpqiu** [comment]
Yes, but it's more convenient for users to choose optimizer by "selecting" the enumerated options instead of "typing" the name.

**Comment by xpqiu** [comment]
there is also a potential plan to implement some customized optimizers.

**Comment by choosewhatulike** [comment]
Rewrite a optimizer class may be redundant. In Tester or Trainer, we actually use optimizers in Pytorch. And further implemented optimizer can inherit `class Optimizer` from Pytorch.

But I agree that choose optimizer should avoid tying names, maybe we can change optimizer.py to import all of optimizers and implemented customized optimizer.
``` python
from torch.optim import *
from XXoptim import XXoptimizer
...
```
And users can "selecting" different optimizer now.

**Comment by xpqiu** [comment]
agree

**Comment by FengZiYjun** [comment]
all right
